api/nodes/views.py

```diff
@@ -632,6 +632,21 @@ def get_queryset(self):
                 .values_list('child__pk', flat=True)
         return self.get_queryset_from_request().filter(pk__in=node_pks).can_view(auth.user).order_by('-modified')
 
+    def get_serializer_context(self):
+        context = super(NodeChildrenList, self).get_serializer_context()
+        region__id = self.request.query_params.get('region', None)
+        id = None
+        if region__id:
+            try:
+                id = Region.objects.get(_id=region__id).id
+            except Region.DoesNotExist:
+                raise InvalidQueryStringError('Region {} is invalid.'.format(region__id))
+
+        context.update({
+            'region_id': id
```

Comment:
Yes this is necessary because components are created via `NodeChildrenList` not `NodeList`. The original add region back-end ticket forgot about this, so I had to add it. This should probably be a mixin that `NodeChildrenList` not `NodeList` share, but I didn't want to add more backend code to what was already a big PR.

This could have definitely been a "comment your own PR" case.

Reply:
Ah, I understand now. Thanks for clarifying.